### PR TITLE
:bug: Add taskgroup when uploading binary

### PR DESCRIPTION
**Pull request by ibolton336**
- Improve handling of taskgroup state by using contextAPI
- Async create the taskgroup before uploading the custom binary in the single app use case. This is currently broken because we require a task group to exist before navigating to the upload file screen. This regression was caused by the move away from useEffect here https://github.com/konveyor/tackle2-ui/pull/1267
Resolves https://issues.redhat.com/browse/MTA-1175

### CODE REVIEW

File: client/src/app/pages/applications/analysis-wizard/custom-rules.tsx [hunk 1]
```diff
@@ -92,7 +92,7 @@ export const CustomRules: React.FC<CustomRulesProps> = (props) => {
     successfullyReadFileCount,
     handleFile,
     removeFiles,
-  } = useRuleFiles(props?.taskgroupID, values.customRulesFiles);
+  } = useRuleFiles(taskGroup?.id, values.customRulesFiles);
```
Comment (sjd78) [inline client/src/app/pages/applications/analysis-wizard/custom-rules.tsx:95]: This is the only bit that looks suspicious to me under brain compile.

What happens if the taskGroup changes?   Maybe something like from `undefined` to `{ id: 5 }`?

I'll run through some smoke tests in a bit.  Maybe that will clear up my confusion.

Comment (ibolton336) [reply client/src/app/pages/applications/analysis-wizard/custom-rules.tsx:95]: +1 Fair concern. The taskgroup should always exist by the time the user arrives at the custom-rules step as it is created either 1) when [onMove](https://github.com/konveyor/tackle2-ui/blob/435b73afc0d3f4500805c618756d49acf445c29b/client/src/app/pages/applications/analysis-wizard/analysis-wizard.tsx#L310) to step 2 of the wizard or 2) user [uploads](https://github.com/konveyor/tackle2-ui/pull/1280/files#diff-defb41a56081c3324873ae6267336b4b17a1cfd59b3f626243cb5565c8bba3c6R119) a binary on step 1 of the wizard.

Comment (sjd78) [reply client/src/app/pages/applications/analysis-wizard/custom-rules.tsx:95]: https://react.dev/learn/reusing-logic-with-custom-hooks#passing-reactive-values-between-hooks

Ok so I'm not concerned about that anymore.

**Review by sjd78 — APPROVED** [review]
LGTM

**Pull request merged**